### PR TITLE
[feature-dropdown] Make dropdown scroll when it doesn't have room

### PR DESCRIPTION
**Pull request by wishdasher**
Best way to observe this behavior is to look at the Multi select with many options example at https://deploy-preview-272--wonder-blocks.netlify.com/#multiselect

Basic rules:
* Dropdown prefers to be under the opener
* If there is not enough room under the opener but enough above, the dropdown appears above the opener.
* The dropdown has a scrollbar if there's not enough room on either side of the opener. This is done by calculating how much space there is and setting a `maxHeight` on the dropdown itself.

Update: 2018 August 24
The current issue happens when there isn't enough room on either side to fit the entire dropdown without a scrollbar. What I believe is happening is: say the dropdown is below the opener and you scroll a bit. `react-popper` does a recalculation and detects that there isn't enough room below the opener so it moves the dropdown to above the opener. Scroll a bit more, and the dropdown gets moved below the opener because of course there was never enough room above the opener either. Repeat ad infinitum.

I haven't really looked into the `popper.js` source, but perhaps one way to improve this is to only allow a flip once one side _gains_ more space than the other. Also worth looking into this function you can supply (https://popper.js.org/popper-documentation.html#modifiers..flip.fn) which is called on each recalculation but I was unable to get this to work.

Related: `https://github.com/FezVrasta/popper.js/issues/550`

### CODE REVIEW

**Comment by jeresig** [comment]
Deploy preview for *wonder-blocks* ready!

Built with commit 81dadc83e3da4a26587dd11dc3689c5348eb0751

https://deploy-preview-272--wonder-blocks.netlify.com

**Comment by wishdasher** [comment]
@kevinbarabash Bad wording on my part -- thanks for pointing it out!

File: packages/wonder-blocks-dropdown/components/dropdown.js [hunk 1]
```diff
@@ -287,7 +312,19 @@ export default class Dropdown extends React.Component<DropdownProps> {
...
+                            escapeWithReference: !moreTopSpace,
+                        },
+                        flip: {
+                            padding: 20,
+                            // We are disabling flipping when there's more
+                            // bottom space because if there are many items,
+                            // react-popper detects that the popper takes up
+                            // more space than available and starts off the
+                            // popper above the opener.
+                            enabled: moreTopSpace,
```
[the remark this answers is not in the document]
Comment (kevinbarabash) [reply packages/wonder-blocks-dropdown/components/dropdown.js]: I was seeing the same thing on Safari.  Thanks for posting a gif of it in action.

[the remark this answers is not in the document]
Comment (wishdasher) [reply packages/wonder-blocks-dropdown/components/dropdown.js]: That's the "wonky behavior" I talked about in the description and is the thing I'm blocked on. I think react-popper with each scroll is like "oh hey, there's not enough room for this on THIS side, let's flip it to the other side!" Currently trying to play with the internal ordering of their update method.

[the remark this answers is not in the document]
Comment (kevinbarabash) [reply packages/wonder-blocks-dropdown/components/dropdown.js]: Hmm.... might have to fix this in `react-popper` itself.

**Comment by wishdasher** [comment]
Not sure how I keep on accidentally closing these

**Comment by wishdasher** [comment]
I didn't close this 😱what is happening

**Comment by wishdasher** [comment]
Updated the PR description to describe the current issue.

**Comment by kevinbarabash** [comment]
I'm going to close this for now.  We can re-open this in the future if we make time to work on it.

**Pull request closed without merging**